source/Xunit.Gherkin.Quick.UnitTests/GherkinScenarioOutlineTests.cs

```diff
@@ -307,12 +307,578 @@ public void ApplyExampleRow_Digests_Row_Values_Into_Scenario_With_DocString_In_S
+            }
+
+            //arrange.
+            var sut = new Gherkin.Ast.ScenarioOutline(
+                null,
+                null,
+                null,
+                "outline123",
+                null,
+                new Gherkin.Ast.Step[]
+                {
+                    new Gherkin.Ast.Step(null, "Given", givenExpression, null),
+                    new Gherkin.Ast.Step(null, "When", "I press check", null),
+                    new Gherkin.Ast.Step(null, "Then", thenExpression, null),
+                },
+                new Gherkin.Ast.Examples[]
+                {
+                    new Gherkin.Ast.Examples(
+                        null,
+                        null,
+                        null,
+                        "existing example",
+                        null,
+                        new Gherkin.Ast.TableRow(null, new Gherkin.Ast.TableCell[]
+                        {
+                            new Gherkin.Ast.TableCell(null, "a"),
+                            new Gherkin.Ast.TableCell(null, "b"),
+                            new Gherkin.Ast.TableCell(null, "result"),
+                        }),
+                        new Gherkin.Ast.TableRow[]
+                        {
+                            new Gherkin.Ast.TableRow(null, new Gherkin.Ast.TableCell[]
+                            {
+                                new Gherkin.Ast.TableCell(null, "5"),
+                                new Gherkin.Ast.TableCell(null, "6"),
+                                new Gherkin.Ast.TableCell(null, "True"),
+                            }),
+                            new Gherkin.Ast.TableRow(null, new Gherkin.Ast.TableCell[]
+                            {
+                                new Gherkin.Ast.TableCell(null, "-5"),
+                                new Gherkin.Ast.TableCell(null, "-6"),
+                                new Gherkin.Ast.TableCell(null, "False"),
+                            }),
+                        })
+                });
+
+            //act.
+            var examplesScenario0 = sut.ApplyExampleRow("existing example", 0);
+            var examplesScenario1 = sut.ApplyExampleRow("existing example", 1);
+
+            //assert.
+            ValidateScenario(
+                examplesScenario0,
+                sut,
+                new string[]
+                {
+                    givenResult1,
+                    "I press check",
+                    thenResult1,
+                });
+
+            ValidateScenario(
+                examplesScenario1,
+                sut,
+                new string[]
+                {
+                    givenResult2,
+                    "I press check",
+                    thenResult2,
+                });
+        }
+
+        [Fact]
+        public void ApplyExampleRow_Digests_Row_Values_Into_Scenario_With_Special_Chars_In_The_Middle_Multiple_Steps()
```

Comment:
Some more complexities with several steps, easier to make it as a Fact rather than integrate into other Theories.